### DISCUSSION TITLE
why cropforeground can

### DISCUSSION THREAD

**Discussion by lg2578** [Project-MONAI/MONAI#6092] [Q&A]
why cropforeground can't crop the top rows  with all values==0
my code:
```
nii=nib.load(files[minidx[1]])
nii=nii.get_fdata()

scale=ScaleIntensity()
def isforeground(x):
    return x>1
print(nii.shape)
cropforeground=CropForeground(isforeground)
# gaussiannoise=RandGaussianNoise(prob=1.0)
# nii=gaussiannoise(nii)
nii=cropforeground(nii[:,:,32])
print(nii.shape)
# nii=scale(nii)
plt.figure(figsize=(10,10))
plt.imshow(nii,cmap='gray')
print(nii[1,:])

```
### outputs:
(266, 266, 144)
torch.Size([266, 254])
tensor([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       dtype=torch.float64)

**Comment by KumoLiu** [comment]
Hi @lg2578, could you please try to add a channel dim in the image? You could use `LoadImage` to load the image.
```
train_transforms = Compose(
    [
        LoadImaged(keys=["img", "seg"], ensure_channel_first=True),
       .......
    ]
)
```

Hope it can help you, thanks!
